### ISSUE TITLE
Add Support for serial port as alternative to `ssh`

### ISSUE THREAD

**Issue by ScheererJ** [gardener/gardenctl-v2#336]
**What would you like to be added**:
`gardenctl` should provide a convenient way to connect to a virtual machine in the cluster node network.

**Why is this needed**:
 Some cluster owners may configure their networks in a way that `ssh` is impossible as ingress traffic may be blocked. Most infrastructures support a serial port/console as alternative means to connect to a virtual machine. As the network path goes through the infrastructure it may be able to reach into such highly isolated clusters.
Serial port/console may require a bastion with a provisioned user/password. However, it does not require a virtual machine to be reachable through the internet.

**Comment by ScheererJ** [comment]
Unfortunately, I cannot provide details why my `ssh` attempts failed. Cluster owners control their network and have plenty options to prevent the connection from working.
In my scenarios, it was likely that internet traffic was routed through peered networks with some firewall dropping packets.

I used https://cloud.google.com/compute/docs/troubleshooting/troubleshooting-using-serial-console, which is fairly easy to utilise. Azure supports something similar (https://learn.microsoft.com/en-us/troubleshoot/azure/virtual-machines/serial-console-overview), but @kon-angelo mentioned that it is a bit more complex to setup. I have not experimented with serial console support on AWS, but I suppose https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-serial-console.html is what I was looking for. It seems different from what you tried as a serial console explicitly does not need network access. Therefore, specifying a subnet seems strange. Nevertheless, it might require additional permissions.

**Comment by petersutter** [comment]
> Nevertheless, it might require additional permissions

yes, this command requires `ec2-instance-connect:SendSerialConsoleSSHPublicKey` permission.
```bash
aws ec2-instance-connect send-serial-console-ssh-public-key --instance-id i-0xxxxxxxxx --serial-port 0 --ssh-public-key file://$TMPDIR/my.id_rsa
```

Then the following should work 
```bash
ssh -i $TMPDIR/my.id_rsa.pub i-0xxxxxxxxx.port0@serial-console.ec2-instance-connect.<region>.aws
```
